src/deluge/deluge.cpp

```diff
@@ -446,6 +449,37 @@ void setupBlankSong() {
 	AudioEngine::mustUpdateReverbParamsBeforeNextRender = true;
 }
 
+/// Can only happen after settings, which includes default settings, have been read
+void setupStartupSong() {
+	auto templatePath = "SONGS/DEFAULT.XML";
+	auto startupSongMode = FlashStorage::defaultStartupSongMode;
+
+	if (startupSongMode == StartupSongMode::BLANK) {
+		setupBlankSong();
+	}
+	else if (startupSongMode == StartupSongMode::TEMPLATE && !storageManager.fileExists(templatePath)) {
+		setupBlankSong();
+		currentSong->writeTemplateSong(templatePath);
+	}
+	else {
+		void* songMemory = GeneralMemoryAllocator::get().allocMaxSpeed(sizeof(Song));
+		currentSong = new (songMemory) Song();
+		auto filename =
+		    startupSongMode == StartupSongMode::TEMPLATE ? "DEFAULT" : runtimeFeatureSettings.getStartupSong();
+		currentSong->setSongFullPath(filename);
+		if (openUI(&loadSongUI)) {
+			loadSongUI.performLoad();
+			if (startupSongMode == StartupSongMode::TEMPLATE) {
+				// Wipe the name so the Save action asks you for a new new
+				currentSong->name.clear();
```

Comment:
The comment says "new new" - did you mean "new name"?

Reply:
updated, thx sean